src/main/java/io/fabric8/jenkins/openshiftsync/BuildWatcher.java

```diff
@@ -163,6 +185,10 @@ public int compare(Build b1, Build b2) {
                             .inNamespace(namespace).withName(buildConfigName)
                             .get();
                     if (bc == null) {
+                        logger.info("skipping listed new build "
+                                + b.getMetadata().getName()
+                                + " no bc at this time");
+                        addBuildToNoBCList(b);
```

Comment:
for the case where we see a build event for a pipeline build, but there is no BC in openshift for it, i think we should ignore it entirely.  The BC isn't going to be created later (someone manually created the build object), so we don't want to sit on that object forever.

Reply:
it is an actual client get of the BC just above, isn't it

and with my testing I believe it has been consistently hitting the `" no job at this time"` block, so that lines up

ok, will change in a moment ...